executor/batch_checker.go

```diff
@@ -48,12 +50,27 @@ type batchChecker struct {
 	dupOldRowValues map[string][]byte
 }
 
-// batchGetOldValues gets the values of storage in batch.
-func (b *batchChecker) batchGetOldValues(ctx sessionctx.Context, t table.Table, handles []int64) error {
-	batchKeys := make([]kv.Key, 0, len(handles))
-	for _, handle := range handles {
-		batchKeys = append(batchKeys, t.RecordKey(handle))
+func (b *batchChecker) RemoveRow(ctx sessionctx.Context, t table.Table, handle int64, row []types.Datum) error {
```

Comment:
Seems only used in `Replace`, why extract this function to `batchChecker` instead of `ReplaceExec`?

Reply:
I have planned to wrap a new API layer around `batchChecker`,  RemoveRow/AddRow/UpdateRow ...
The benefit of that API layer is that we won't be trapped by this kind of invoking:

```
// blah blah ...
table.RemoveRecord() // modify table
// forget to backfill batchChecker indices...
```

But I found it's too much changes to accomplish in this single PR.

Should I reset this change, or just keep it here and add other API later? @coocood

Reply:
I think reset the change is better.